src/core/catalog.js

```diff
@@ -1400,25 +1401,10 @@ class Catalog {
           }
 
           if (js) {
-            // Attempt to recover valid URLs from `JS` entries with certain
-            // white-listed formats:
-            //  - window.open('http://example.com')
-            //  - app.launchURL('http://example.com', true)
-            const URL_OPEN_METHODS = ["app.launchURL", "window.open"];
-            const regex = new RegExp(
-              "^\\s*(" +
-                URL_OPEN_METHODS.join("|").split(".").join("\\.") +
-                ")\\((?:'|\")([^'\"]*)(?:'|\")(?:,\\s*(\\w+)\\)|\\))",
-              "i"
-            );
-
-            const jsUrl = regex.exec(stringToPDFString(js));
-            if (jsUrl && jsUrl[2]) {
-              url = jsUrl[2];
-
-              if (jsUrl[3] === "true" && jsUrl[1] === "app.launchURL") {
-                resultObj.newWindow = true;
-              }
+            const parsed = getURL(stringToPDFString(js));
+            url = parsed.url;
+            resultObj.newWindow = parsed.newWindow;
+            if (url) {
               break;
             }
```

Comment:
With the other suggestions implementation, this could be simplified a bit more:
```javascript
const jsURL = js && recoverJsURL(stringToPDFString(js));
if (jsURL) {
  url = jsURL.url;
  resultObj.newWindow = jsURL.newWindow;
  break;
}
```